### PR TITLE
Add BarButton

### PR DESCRIPTION
**Pull request by gregorylegarec**
The BarButton is targeted to be injected into CozyBar.

A first usage is to offer a back button on mobile.

```jsx
const { BarLeft } = cozy.bar
return (
  <BarLeft>
    <BaButton icon="back" />
  </BarLeft>
)
```

[Styleguidist](https://gregorylegarec.github.io/cozy-ui/react/#!/BarBackButton)

### CODE REVIEW

File: react/BarBackButton/index.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  render() {
+    const { disabled, href, onClick } = this.props
+    return (
+      <MaybeLink
+        className={styles['barleft']}
+        href={!disabled && href}
+        onClick={!disabled && onClick}
+      >
+        <Icon icon="back" className={disabled ? 'u-silver' : 'u-coolGrey'} />
+      </MaybeLink>
```
Comment (CPatchane) [inline react/BarBackButton/index.jsx]: It looks like a special icon button to me but it seems not specific to the bar, so I am not comfortable with the naming `BarBackButton`

Comment (ptbrowne) [reply react/BarBackButton/index.jsx]: The u-silver, u-coolgrey behavior is specific to BarBackButton. Do you see another instance in Cozy that has this behavior ? I prefer not to abstract a behavior early here. I don't see what we would gain.

Comment (ptbrowne) [reply react/BarBackButton/index.jsx]: Maybe BarButton instead of IconButton ?

<BarButton icon='back' />
<BarButton icon='sandwich' />

Comment (CPatchane) [reply react/BarBackButton/index.jsx]: At least `BarButton` is still least specific than `BarBackButton` so 👍

Comment (gregorylegarec) [reply react/BarBackButton/index.jsx]: So let's go for a `<BackButtion />` I assume that `icon` should now be a prop, or should we wait for the express need ?

Comment (CPatchane) [reply react/BarBackButton/index.jsx]: It should be a prop like in @ptbrowne example

**Comment by gregorylegarec** [comment]
Updated ✅

File: react/BarButton/Readme.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+```jsx static
+  const { BarLeft } = cozy.bar
+  return (
+    <BarLeft>
+      <BarButton />
+    </BarLeft>
+    )
+```
+
+## Props
```
Comment (y-lohse) [inline react/BarButton/Readme.md]: Note that you can document the prop directly with comments https://react-styleguidist.js.org/docs/documenting.html#code-comments-and-proptypes

Comment (ptbrowne) [reply react/BarButton/Readme.md]: Yes please.

File: react/BarButton/Readme.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+<BarButton icon="cube" />
+```
+
+### `onClick`
+
+Add an `onClick` event handler on the component.
+
+```jsx
+<BarButton icon="gear" onClick={() => alert('BarButton has been clicked')} />
+```
```
Comment (ptbrowne) [inline react/BarButton/Readme.md]: I think we should only document via Proptypes.

File: react/BarButton/Readme.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+Used to render the internal `<Icon />` component.
+
+```jsx
+<BarButton icon="cube" />
+```
+
+### `onClick`
+
+Add an `onClick` event handler on the component.
```
Comment (ptbrowne) [inline react/BarButton/Readme.md]:
```suggestion
Adds an `onClick` event handler on the component.
```

File: react/BarButton/Readme.md (hunk 3 above)
Comment (gregorylegarec) [reply react/BarButton/Readme.md]: Don't you think it's pretty convenient to have those examples rendered in styleguidist to test them manually ?

It allows us to get the component snapshoted by Jest as well.

Comment (ptbrowne) [reply react/BarButton/Readme.md]: > It allows us to get the component snapshoted by Jest as well.

You're right, I thought that with `jsx` we had only the code and not the component displayed. I forgot that it was `static` that was doing this.

**Pull request merged**